packages/main/src/components/Form/index.tsx

```diff
@@ -0,0 +1,88 @@
+import { Grid } from '@ui5/webcomponents-react/lib/Grid';
+import React, { FC, forwardRef, ReactElement, ReactNode, ReactNodeArray, Ref, useMemo } from 'react';
+import { CommonProps } from '../../interfaces/CommonProps';
+import { Title } from '@ui5/webcomponents-react/lib/Title';
+import { TitleLevel } from '@ui5/webcomponents-react/lib/TitleLevel';
+import styles from './Form.jss';
+import { createUseStyles } from 'react-jss';
+import { useViewportRange } from '../../../../base/src/hooks/useViewportRange';
+import { FormGroup } from './FormGroup';
+import CurrentRange from './CurrentViewportRange';
+import { JSSTheme } from '../../interfaces/JSSTheme';
+
+export interface FormPropTypes extends CommonProps {
+  /**
+   * Components that are placed into Form.
+   */
+  children: ReactNode | ReactNodeArray;
+  /**
+   * Form title
+   */
+  title?: string;
+}
+
+let formGroups, updatedTitle;
+const useStyles = createUseStyles<JSSTheme, keyof ReturnType<typeof styles>>(styles, { name: 'Form' });
+
+const Form: FC<FormPropTypes> = forwardRef((props: FormPropTypes, ref: Ref<HTMLDivElement>) => {
+  const { title, children } = props;
+
+  const classes = useStyles();
+  const formTitleStyle = useStyles(classes).formTitle;
+  const formTitlePadding = useStyles(classes).formTitlePaddingBottom;
+
+  const [currentRange] = useViewportRange();
+
+  useMemo(() => {
```

Comment:
you should make the useMemo return your formGroups and updatedTitle.
`const [formGroups, updatedTitle] = useMemo(() => {
    ...
   return [formGroups, updatedTitle];
})`